tracer/test/Datadog.Trace.TestHelpers/SkipOn.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="SkipOn.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Datadog.Trace.TestHelpers;
+
+/// <summary>
+/// SkipOn helper
+/// </summary>
+public static class SkipOn
+{
+    /// <summary>
+    /// Platform enum
+    /// </summary>
+    public enum Platform
+    {
+        /// <summary>
+        /// Windows platform
+        /// </summary>
+        Windows,
+
+        /// <summary>
+        /// Linux platform
+        /// </summary>
+        Linux,
+
+        /// <summary>
+        /// MacOs platform
+        /// </summary>
+        MacOs
+    }
+
+    /// <summary>
+    /// Architecture enum
+    /// </summary>
+    public enum Architecture
+    {
+        /// <summary>
+        /// X86 arch
+        /// </summary>
+        X86,
+
+        /// <summary>
+        /// X64 arch
+        /// </summary>
+        X64,
+
+        /// <summary>
+        /// ARM64 arch
+        /// </summary>
+        ARM64
+    }
+
+    public static void PlatformAndArchitecture(Platform platform, Architecture architecture)
```

Comment:
I like this. Rather than adding `if (EnvironmentTools.IsOsx())` to the other tests, might be nice to add another overload here too: 
```suggestion
    public static void Platform(Platform platform)
    {
        // ...
    }
    
    public static void PlatformAndArchitecture(Platform platform, Architecture architecture)
```

So the tests have a nice 
`SkipOn.Platform(OSPlatform.OSX)`

Reply:
done in [7f42b05](https://github.com/DataDog/dd-trace-dotnet/pull/3863/commits/7f42b05e794ac84b30e5fc059d4d6c8c7b8eb4af)